syntaxes/connectors.jsonselection.yaml

```diff
@@ -0,0 +1,270 @@
+# use 'js-yaml' to convert to json
+# npx js-yaml syntaxes/connectors.jsonselection.yaml >| syntaxes/connectors.jsonselection.json
+name: "Apollo Connectors JSONSelection"
+scopeName: "source.apollo.jsonselection"
+patterns:
+  - include: "#JSONSelection"
+  - include: "source.graphql#graphql-skip-newlines"
+repository:
+  # JSONSelection        ::= PathSelection | NamedSelection*
+  JSONSelection:
+    name: "meta.JSONSelection"
+    patterns:
+      - include: "#PathSelection"
+      - include: "#NamedSelection"
+      - include: "#Comment"
+  # SubSelection         ::= "{" NamedSelection* "}"
+  SubSelection:
+    name: "meta.SubSelection"
+    begin: "\\s*({)"
+    beginCaptures:
+      "1":
+        name: "punctuation.SubSelection.open"
+    end: "\\s*(})"
+    endCaptures:
+      "1":
+        name: "punctuation.SubSelection.close"
+    patterns:
+      - include: "#NamedSelection"
+      - include: "#Comment"
+  # NamedSelection       ::= NamedPathSelection | PathWithSubSelection | NamedFieldSelection | NamedGroupSelection
+  NamedSelection:
+    name: "meta.NamedSelection"
+    patterns:
+      - include: "#PathWithSubSelection"
+      - include: "#NamedFieldSelection"
+      - include: "#NamedPathSelection"
+      - include: "#NamedGroupSelection"
+  # NamedPathSelection   ::= Alias PathSelection
+  NamedPathSelection:
+    begin: "\\b([a-zA-Z][0-9a-zA-Z_]*)\\s*(:)"
+    beginCaptures:
+      "1":
+        name: "string.unquoted.NamedPathSelection.alias"
```

Comment:
So can each of these sections get their own colors? I'd love to have one color for the "output" name (alias in this case) and another for things that are inputs.

Reply:
Yes, they can!

Open a text editor and execute the "Developer: inspect editor tokens and scopes" command

![image](https://github.com/user-attachments/assets/9c11445e-65cc-4b8a-ae05-742de22f9da1)

that will give you this pop-up:

![image](https://github.com/user-attachments/assets/c03c1339-1d9a-4d6d-9695-aed9542d1df2)


And then you can move your cursor around and see what tokens things in other languages have, as an inspiration.

---

![image](https://github.com/user-attachments/assets/c3af9462-8e1a-45e6-ac59-84d782649d49)
![image](https://github.com/user-attachments/assets/b8403166-dc27-438d-86a4-51e2da416e18)

You see that GraphQL uses `string.unquoted` and `variable` as highlighting scopes here, but I also see that the `alias` is never applied because `variable.KeyPath.key.identifier` seems to have a higher priority somehow.

I'll debug that :)

Reply:
So... with some changes:

![image](https://github.com/user-attachments/assets/2f83e00b-f560-455c-b51f-5c7a79c3a61a)

Reply:
Amazing! A couple more notes on what we might want to add in future iterations:

1. `$` and `@` as well as an identifier following $, like `$args`, `$this`, etc. are sort of our keywords? They're special, at very least, and might be worth highlighting separately.
2. Anything in `$()` or as an argument to a method is sort of a different language mode. We call them literals, but they can also have some expressions within them. So `${ blah: $.blah }` and `$({ blah: $.blah })` are both valid, and look very similar, but do different things. It'd be nice to highlight that difference somehow (effectively, just the presence of the parens causes it).

Reply:
I can configure `{(` and `)}` as a special kind of bracket pairs (this is VSCode, not TextMate!) so they will always have the same color and highlight in pairs together:
![image](https://github.com/user-attachments/assets/c6d49827-a451-410b-9da8-b9175f972975)

Reply:
that's cool! i don't think it's exactly what we want tho. the way i see it, when we're inside parens (for both literal values like `$(...)` or for method arguments like `->method(...)`), we're now in "literal value" mode instead of "selection" mode. So I'd color the outer parens as one thing, and any inner values to indicate that they're different from selections. i dunno exactly what that looks like

Reply:
As for 1., does this work for you @dylan-apollo ?
![image](https://github.com/user-attachments/assets/9fb17a95-52af-4178-abb7-362d45115ad9)

Reply:
@lennyburdette then I'm undoing that change and we go back to this:
![image](https://github.com/user-attachments/assets/97f50d89-1740-4547-b840-cc40bc2dede6)


Honestly I don't know if I can do a lot to visually highlight that difference more, apart from adding a full other "sub-language", which I'd rather avoid as this would become even more unmaintainable 😅

Reply:
> Maybe we just color parens differently than {} for now, since parens always represent entering literal mode? We could do something special with semantic tokens or something if we really want in the future

We'll need to test that out in the non-VSCode editors. VSCode pretty much ignores everything else and applies its "bracket pair coloring" logic over the top of configured bracket pairs (which we want to have so it highlights start and end).

Reply:
Then maybe it's just fine 😅. Certainly shouldn't block getting the highlighting of the rest of the language out there